pydantic/typing.py

```diff
@@ -121,12 +121,9 @@ def get_args(tp: Type[Any]) -> Tuple[Any, ...]:
             get_args(Union[int, Tuple[T, int]][str]) == (int, Tuple[str, int])
             get_args(Callable[[], T][int]) == ([], int)
         """
-        try:
-            args = typing_get_args(tp)
-        except IndexError:
-            args = ()
+        args = typing_get_args(tp)
         # the fallback is needed for the same reasons as `get_origin` (see above)
-        return args or getattr(tp, '__args__', ()) or generic_get_args(tp)
+        return tuple(args or getattr(tp, '__args__', ()) or generic_get_args(tp))
```

Comment:
This was need for the `conlist(str)` case in tests which was previously returning a string.